tools/ppa-upload.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -e
+
+if [ -z "${RELEASE}" ]; then
+  echo "ERROR: RELEASE environment variable needs to be set to the" >&2
+  echo "  target Ubuntu version." >&2
+  exit 1
+fi
+
+if ! [[ "${TARGET}" =~ ^dev|rc|release$ ]]; then
+  echo "ERROR: TARGET environment variable needs to be set to the" >&2
+  echo "  target name - one of dev, rc, release. " >&2
+  exit 3
+fi
+
+source <( python <<EOF
+import os
+from launchpadlib.launchpad import Launchpad
+
+lp = Launchpad.login_anonymously("mir-ci",
+                                 "production",
+                                 "/tmp/lplib",
+                                 version="devel")
+
+ubuntu = lp.distributions["ubuntu"]
+series = ubuntu.getSeries(name_or_version=os.environ['RELEASE'])
+
+print("UBUNTU_SERIES={}".format(series.name))
+print("UBUNTU_VERSION={}".format(series.version))
+EOF
+)
+
+if [ -z "${UBUNTU_SERIES}" -o -z "${UBUNTU_VERSION}" ]; then
+  echo "ERROR: \"${RELEASE}\" was not recognized as a valid Ubuntu series" >&2
+  exit 2
+fi
+
+GIT_REVISION=$( git rev-parse --short HEAD )
+
+# TODO target release
+if [ "${TARGET}" = "release" ]; then
+  echo "ERROR: \"release\" target not supported yet." >&2
+  exit 4
+elif [ "${TARGET}" = "rc" ]; then
+  # we're in a release pull request
+  if [[ "$( git describe --exact-match )" =~ v[0-9.]+ ]] 2> /dev/null; then
+    # this is a final release, use the tag version
+    MIR_VERSION=$( git describe | sed 's/^v//' )
+  else
+    # determine the patch release
+    if ! [[ "${TRAVIS_PULL_REQUEST_BRANCH}" =~ ^release/([0-9\.]+) ]]; then
+      echo "ERROR: Can't target rc outside of a release Pull Request" >&2
+      exit 5
+    fi
+    # find the last tagged patch version
+    PATCH_VERSION=$( git describe --abbrev 0 --match "v${BASH_REMATCH[1]}*" \
+                     2> /dev/null | sed 's/^v//')
+    if [ -z "${RC_VERSION}" ]; then
+      # start with patch version 0
+      MIR_VERSION=${BASH_REMATCH[1]}.0
+    else
+      # increment the patch version
+      MIR_VERSION=$( echo ${RC_VERSION} | perl -pe 's/^((\d+\.)*)(\d+)$/$1.($3+1)/e' )
+    fi
+
+    # use the number of commits since master
+    GIT_COMMITS=$( git rev-list --count master..HEAD )
+    MIR_VERSION=${MIR_VERSION}~rc-${GIT_COMMITS}-g${GIT_REVISION}
```

Comment:
It would probably be better to use ‘.’ rather than ‘-’ as a separator here - as I think you suspect, ‘-’ is parsed by dpkg to indicate the debian-part vs the upstream-part.

Reply:
Agreed.